src/librustdoc/json/conversions.rs

```diff
@@ -87,59 +87,78 @@ impl JsonRenderer<'_> {
     }
 }
 
+crate trait FromWithTcx<T> {
+    fn from_tcx(f: T, tcx: TyCtxt<'_>) -> Self;
+}
+
+crate trait IntoWithTcx<T> {
+    fn into_tcx(self, tcx: TyCtxt<'_>) -> T;
+}
+
+impl<T, U> IntoWithTcx<U> for T
+where
+    U: FromWithTcx<T>,
+{
+    fn into_tcx(self, tcx: TyCtxt<'_>) -> U {
+        U::from_tcx(self, tcx)
+    }
+}
```

Comment:
Why do these need to be traits? It seems simpler to just use free functions. Otherwise you'll have to redo this all again next time you want to add a parameter.

Reply:
I kept it as close as possible to the current code (which was using `From`). Otherwise I'll have to replace all the current trait implementations with free functions for all different types.

Reply:
Hmm, well I'm not convinced the traits are pulling their weight, but we can fix it in a follow-up to avoid unnecessary churn.